python_modules/dagster/dagster/_core/definitions/asset_definition.py

```diff
@@ -0,0 +1,189 @@
+from typing import Any, Callable, Dict, Iterable, Mapping, Optional, Sequence, Union
+
+from dagster._core.definitions.asset_dep import CoercibleToAssetDep
+from dagster._core.definitions.asset_key import AssetKey, CoercibleToAssetKey
+from dagster._core.definitions.asset_spec import AssetSpec
+from dagster._core.definitions.assets import AssetsDefinition
+from dagster._core.definitions.auto_materialize_policy import AutoMaterializePolicy
+from dagster._core.definitions.backfill_policy import BackfillPolicy
+from dagster._core.definitions.freshness_policy import FreshnessPolicy
+from dagster._core.definitions.node_definition import NodeDefinition
+from dagster._core.definitions.partition import PartitionsDefinition
+from dagster._core.errors import DagsterInvalidDefinitionError
+from dagster._utils.warnings import disable_dagster_warnings
+
+
+class AssetDefinition(AssetsDefinition):
+    def __init__(
+        self,
+        key: CoercibleToAssetKey,
+        *,
+        keys_by_input_name: Mapping[str, AssetKey] = {},
+        output_name: Optional[str] = None,
+        node_def: Optional[NodeDefinition] = None,
+        deps: Iterable[CoercibleToAssetDep],
```

Comment:
An alternative would be to have it accept a `spec: AssetSpec` argument instead of all these properties.